src/gpuarray_reduction.c

```diff
@@ -188,63 +695,101 @@ static void  appendIdxes                        (strb*              s,
 
 /**
  * @brief Check the sanity of the arguments, in agreement with the
- *        documentation for GpuArray_maxandargmax().
+ *        documentation for GpuArray_reduction().
  *
  *        Also initialize certain parts of the context.
  *
  * @return GA_INVALID_ERROR if arguments invalid; GA_NO_ERROR otherwise.
  */
 
-static int   maxandargmaxCheckargs              (maxandargmax_ctx*  ctx){
-	int i;
+static int   reduxCheckargs                (redux_ctx*  ctx){
+	int i, ret;
+	const strb INIT_STRB = STRB_STATIC_INIT;
 
 	/**
 	 * We initialize certain parts of the context.
 	 */
 
-	ctx->ret           = GA_NO_ERROR;
 	ctx->axisList      = NULL;
 	ctx->gpuCtx        = NULL;
 
-	ctx->dstMaxType    = ctx->dstArgmaxType = NULL;
+	ctx->srcTypeStr    = ctx->dstTypeStr    = ctx->dstArgTypeStr =
+	ctx->accTypeStr    = ctx->idxTypeStr    = NULL;
+	ctx->initVal       = NULL;
 	ctx->ndh           = 0;
+	ctx->ndhd          = 0;
+	ctx->ndhr          = 0;
 	ctx->sourceCode    = NULL;
+	ctx->s             = INIT_STRB;
```

Comment:
I think I'll change this to `(strb)STRB_STATIC_INIT`, but it would be lovely if `STRB_STATIC_INIT` incorporated `(strb)` itself.